flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -1600,23 +1619,38 @@ private Object[][] buildTableColumns(ResolvedSchema schema) {
                                                             "PRI(%s)",
                                                             String.join(", ", columns))));
                         });
-
+        boolean nonComments = isSchemaNonColumnComments(schema);
         return schema.getColumns().stream()
                 .map(
                         (c) -> {
                             final LogicalType logicalType = c.getDataType().getLogicalType();
+                            if (nonComments) {
+                                return new Object[] {
```

Comment:
Can we just create `List<Object>`, with comments, we can add a comment.
Then `List.toArray` to a object array.

Reply:
done :)